py3status/modules/net_rate.py

```diff
@@ -172,9 +174,10 @@ def currentSpeed(self):
         elif not interface:
             response["full_text"] = self.format_no_connection
         else:
-            self.py3.threshold_get_color(delta["down"], "down")
-            self.py3.threshold_get_color(delta["total"], "total")
-            self.py3.threshold_get_color(delta["up"], "up")
+            for x in self.thresholds_init:
+                if x in delta:
```

Comment:
I think delta could be None and this would raise an exception right?

Reply:
That's a possibility. I'll look into this.

Reply:
I think it'd be caught by `if hide` or `elif not interface` first.

Reply:
You're right indeed